lib/iris/mesh/components.py

```diff
@@ -2928,6 +3071,10 @@ def copy(self, points=None, bounds=None):
         new_coord = MeshCoord(mesh=self.mesh, location=self.location, axis=self.axis)
         return new_coord
 
+    def rename(self, name: str | None) -> None:
+        """Ensure you that renaming MeshCoords is done via the Coord attached to the Mesh."""
+        raise ValueError("To rename a MeshCoord, you have to go via the attached mesh.")
```

Comment:
A more descriptive error message would be helpful, rather than "... go via the attached mesh." maybe "... rename the coordinate on the mesh from which this derives."